web/auth/jwt.go

```diff
@@ -0,0 +1,49 @@
+package auth
+
+import (
+	"fmt"
+	"log"
+
+	pb "github.com/autograde/quickfeed/ag"
+	"github.com/autograde/quickfeed/database"
+	"github.com/golang-jwt/jwt"
+)
+
+type Claims struct {
+	jwt.StandardClaims
+	UserID  uint64                              `json:"user_id"`
+	Admin   bool                                `json:"admin"`
+	Courses map[uint64]pb.Enrollment_UserStatus `json:"courses"`
+}
+
+type TokenManager struct {
+	TokensToUpdate []uint64
+}
+
+func (tm *TokenManager) JWTUpdateRequired(claims *Claims) bool {
```

Comment:
Would be nice to have proper tests for these methods.

Reply:
Tests added.